waves_gateway/service/coin_transaction_consumer_impl.py

```diff
@@ -119,15 +124,39 @@ def _handle_transaction_receiver(self,
 
         # ---------- Pre-Check ----------
 
+        sender_array = []
+        if senders:
+            for sender in senders:
+                sender_array.append(sender.address)
+
         # transaction must have minimum amount to be handled
         if amount_after_fees <= 0:
-            self._logger.warn("Received transaction %s with an amount of %s, but it is less than the at least "
-                              "required amount. Will be skipped, but marked as processed.", tx, receiver.amount)
+            message = "Received transaction " + tx + " with an amount of " + str(receiver.amount) +", but it is less than the at least " \
+                      "required amount. Will be skipped, but marked as processed."
+            self._logger.warn(message)
+
+            failed_transaction = FailedTransaction("", self._public_configuration.custom_currency_name,
+                                                   "amount too small",
+                                                   str(message).format(), datetime.datetime.now(), {
+                                                       "tx": tx,
+                                                       "receiver": receiver.address,
```

Comment:
Why don't you refactor the method, so that the complete transaction is overhanded? In this case, it would not be necessary to rebuild the transaction.